src/node/txreconciliation.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/txreconciliation.h>
+
+#include <util/system.h>
+
+#include <unordered_map>
+#include <variant>
+
+
+namespace {
+
+/**
+ * Keeps track of reconciliation-related per-peer state.
+ */
+class ReconciliationState
+{
+};
+
+} // namespace
+
+/** Actual implementation for TxReconciliationTracker's data structure. */
+class TxReconciliationTracker::Impl
+{
+    mutable Mutex m_mutex;
+
+    /**
+     * Keeps track of reconciliation states of eligible peers.
+     * For pre-registered peers, the locally generated salt is stored.
+     * For registered peers, the locally generated salt is forgotten, and the state (including
+     * "full" salt) is stored instead.
+     */
+    std::unordered_map<NodeId, std::variant<uint64_t, ReconciliationState>> m_states GUARDED_BY(m_mutex);
+
+public:
+    // Local protocol version
+    // Made public to supress -Wunused-private-field. Should be made private when becomes used.
+    const uint32_t m_recon_version;
+
+    explicit Impl(uint32_t recon_version) : m_recon_version(recon_version) {}
+
+    uint64_t PreRegisterPeer(NodeId peer_id) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+    {
+        LOCK(m_mutex);
+        // We do not support reconciliation salt/version updates.
+        assert(m_states.find(peer_id) == m_states.end());
+
+        LogPrint(BCLog::TXRECON, "Pre-register peer=%d.\n", peer_id);
+        uint64_t local_recon_salt{GetRand(UINT64_MAX)};
```

Comment:
0d7823a20650ef3b5f2c2da6b33f6dd142636c38 suggest `LogPrintLevel` (and const while retouching)

```diff
-        LogPrint(BCLog::TXRECON, "Pre-register peer=%d.\n", peer_id);
-        uint64_t local_recon_salt{GetRand(UINT64_MAX)};
+        LogPrintLevel(BCLog::TXRECON, BCLog::Level::Debug, "%s\n", strError.original);
+        const uint64_t local_recon_salt{GetRand(UINT64_MAX)};
```